.github/workflows/cross-version-tests.yml

```diff
@@ -18,6 +18,10 @@ on:
     branches:
       - master
 
+concurrency:
+  group: cross-version-tests-${{ github.ref }}
+  cancel-in-progress: true
+
```

Comment:
https://github.com/vitessio/vitess/pull/8546 describes well how the concurrency feature work:

> When a new commit is added to a pull request, GitHub Actions will add each workflow to groups. Groups are defined like this: branch_name + workflow_name, thus every workflow of the pull request's branch will be in their own unique group. GitHub Actions concurrency feature uses these groups to cancel workflow when new workflows (due to a new commit being pushed) are added to the queue. If for a given group there is a more recent workflow in the queue, the older workflows of the same group will be canceled by GitHub Actions, and only the most recent one will be executed.